cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/UsageHandler.java

```diff
@@ -55,7 +59,8 @@ public void getAppDatasetUsage(HttpRequest request, HttpResponder responder,
                                  @PathParam("app-id") String appId) {
     final ApplicationId id = new ApplicationId(namespaceId, appId);
     Set<? extends EntityId> ids = registry.getDatasets(id);
```

Comment:
not part of your PR, but why are we doing this? Why not just assign it to `Set<DatasetId>`?